test/testPartyAttachments.js

```diff
@@ -0,0 +1,1025 @@
+ * content-length: 10977
+ * host: localhost:5000
+ * Connection: close
+ *
+ * ----------------------NODENEEDLEHTTPCLIENT
+ * Content-Disposition: form-data; name="body"
+ *
+ * [{"file":"profile.png","attachment":{"key":"18f6f8ea-3926-4fe7-80a0-49cec88a66fd","description":"this is MY file with key 18f6f8ea-3926-4fe7-80a0-49cec88a66fd"},"resource":{"href":"/partiesS3/2691d53a-6f24-416e-9621-3cd14c05c5a6"}}]
+ * ----------------------NODENEEDLEHTTPCLIENT
+ * Content-Disposition: form-data; name="data"; filename="profile.png"
+ * Content-Transfer-Encoding: binary
+ * Content-Type: image/png
+ *
+ * <binary data>
+ * ----------------------NODENEEDLEHTTPCLIENT--
+ * ```
+ *
+ * An alternative form could be using additional headers to indicate file properties instead of the JSON 'body':
+ *
+ * ```js
+ *   const form = new FormData();
+ *   form.append(`1_${localFilename}`, fs.createReadStream(localFilename), { header: { hello: 'goodbye' } });
+ *   form.append(`2_${localFilename}`, fs.createReadStream(localFilename));
+ *   form.append(`3_${localFilename}`, fs.createReadStream(localFilename));
+ * ```
+ *
+ * Which translates to something like this:
+ * ```
+ * POST /partiesS3 HTTP/1.1
+ * content-type: multipart/form-data; boundary=--------------------------732677279853170760492713
+ * Host: localhost:5000
+ * Content-Length: 31895
+ * Connection: close
+ *
+ * ----------------------------732677279853170760492713
+ * Content-Disposition: form-data; name="1_test/orange-boy-icon.png"; filename="orange-boy-icon.png"
+ * Content-Type: image/png
+ * hello: goodbye
+ *
+ * <binary data>
+ * ----------------------------732677279853170760492713
+ * Content-Disposition: form-data; name="2_test/orange-boy-icon.png"; filename="orange-boy-icon.png"
+ * Content-Type: image/png
+ *
+ * <binary data>
+ * ----------------------------732677279853170760492713
+ * Content-Disposition: form-data; name="3_test/orange-boy-icon.png"; filename="orange-boy-icon.png"
+ * Content-Type: image/png
+ *
+ * <binary data>
+ * ----------------------------732677279853170760492713--
+ * ```
+ *
+ * @param {THttpClient} httpClient
+ * @param {string} resourceUrl is the url of the resource for which you want to put an attachment
+ *                              for example https://localhost:5000/partiesS3/<some-guid>
+ * @param {Array<TFileToUpload | TFileToCopy>} fileDetails
+ * @returns {Promise<THttpResponse>} a http response
+ */
+async function doPutFiles(httpClient, resourceUrl, fileDetails) {
+  const body = createUploadBody(fileDetails);
+  const formData = new FormData();
+  formData.append('body', JSON.stringify(body));
+
+  fileDetails
+    .forEach((f) => {
+      if ('localFilename' in f) {
+        const { remotefileName, localFilename, attachmentKey } = f;
+        formData.append(attachmentKey, fs.createReadStream(localFilename), {
+          filename: remotefileName,
+          contentType: 'image/png',
```

Comment:
The content type is added automatically based on the filename (we might use a different content type) so this object can be replaced with `remotefileName` only.